packages/beacon-node/src/chain/validation/block.ts

```diff
@@ -94,7 +94,7 @@ export async function validateGossipBlock(
   // single bad block can trigger
   // Note: Ensure this check is done before calling chain.regen.getBlockSlotStat as this is the function that does various epoch transitions.
   // Note: This validation check is not part of the spec.
-  if (chain.opts.maxSkipSlots != null && parentBlock.slot + chain.opts.maxSkipSlots < blockSlot) {
+  if (chain.opts.maxSkipSlots !== undefined && parentBlock.slot + chain.opts.maxSkipSlots < blockSlot) {
```

Comment:
type of `chain.opts.maxSkipSlots` is `number | undefined`, so the compiler takes care that `null` can never be a value.

The code before the diff when `chain.opts.maxSkipSlots` is `undefined` works because even though `chain.opts.maxSkipSlots != null` returns true (ie undefined != null returns true),  `parentBlock.slot + chain.opts.maxSkipSlots < blockSlot` will always return false because `parentBlock.slot + undefined < blockSlot` will be false.

The better way is to perform strict check against `undefined`, this ensures the if check short-circuits on first check.